package.json

```
@@ -11,11 +11,11 @@
     "type-check": "tsc --noEmit",
```

```suggestion
    "smoketest": "yarn type-check && yarn eslint && yarn prettier:check && yarn build",
    "type-check": "tsc --noEmit",
```

How about having a `smoketest` command? I can also change the suggestion so that it runs the `prettier:fix` and `eslint:fix` commands instead of just linting for bad stuff.

The workflow could then be:

1. `yarn dev` work on stuff
2. `yarn smoketest` to check and/or fix linting errors
3. `yarn smoketest` passes -> push to `main` / open PR

---

> I prefer not to fix anything in a pipeline, so there are no changes committed in CI/CD.
> 
> I'm not really sure when would this `smoketest` command be executed?

I don't intend to use the `smoketest` command in the CI/CD. 😉 
This command is purely for local development, and to help new contributors perform all the wanted checks locally before making a pull request.

That way we could have a new `contributing.md` file to help new contributors get started, and just say: if `smoketest` passes, you're good to go make a PR. This just simplifies the on-boarding process for new users, and takes the guesswork out of "have I done everything I needed before making the PR/pushing to `main`?".

---

Ooh, I see. Sorry I understood you wanted to use it for CI/CD as this PR is modifying GitHub Actions. It sounds like a good idea, we can do it in a separate PR as you mentioned.